app/models/endpoint.rb

```diff
@@ -16,7 +16,31 @@ class Endpoint < ApplicationRecord
   validates :endpoint_type, presence: true
   validates :endpoint_node, presence: true
   validates :storage_location, presence: true
-  validates :recovery_cost, presence: true
+
+  scope :archive, lambda {
+    # TODO: maybe endpoint_class should be an enum or a constant?
```

Comment:
Agree.  That would let us do `Endpoint_type.archive` as an auto-magic activerecord scope.

Reply:
Created #797 re: this.  Put it in backlog.